kotlinx-coroutines-core/jvm/src/internal/StackTraceRecovery.kt

```diff
@@ -38,7 +38,8 @@ private fun <E : Throwable> E.sanitizeStackTrace(): E {
     return this
 }
 
-internal actual fun <E : Throwable> recoverStackTrace(exception: E, continuation: Continuation<*>): E {
+@Suppress("NOTHING_TO_INLINE") // Inline for better R8 optimization
+internal actual inline fun <E : Throwable> recoverStackTrace(exception: E, continuation: Continuation<*>): E {
```

Comment:
Changes in this file should blow up bytecode size.
`inline` on `recoveryDisabled` should be enough, otherwise let's delegate it to R8

Reply:
It has to be inline, or otherwise the `StrackTraceRecoveryKt.class` cannot be removed by R8. It is correctly removed now, which is verified by `-checkdiscard class kotlinx.coroutines.internal.StackTraceRecoveryKt`. 

Notice how `recoverStackTrace` function body checks for `recoveryDisabled` and the calls the actual `recoverFromStackFrame` implementation function. This way, when it is inlined and `recoveryDisabled` is also inline (check it out!) and statically knows that recovery is disabled, then no references to  `StrackTraceRecoveryKt.class` are left and it is removed completely.

Reply:
If `recoveryDisabled` is inline and statically known, optimizer should make `recoverStackTrace` a no-op and then safely remove.
Even if not (let's report it then), the only thing that should be left is empty `StrackTraceRecoveryKt` class that is not a big deal.

Reply:
It does not work this way. It makes it no-op, but does not remove, because it assumes that there might be class-initialization side-effects on calling this method. Removing it would require a separate `assumenosideeffects` to R8. Keeping `StrackTraceRecoveryKt` is a big deal, though, because it also contains various statics and their initialization. We have to make sure it is totally gone.